### ISSUE TITLE
npm start not working on fresh Vagrant setup

### ISSUE THREAD

**Issue by SergeantFTC** [HabitRPG/habitica#3150]
Here's the results of running "npm start" on the Vagrant environment I just got set up as per the latest instructions on the wiki: 

vagrant@habitrpg:~$ cd /vagrant
vagrant@habitrpg:/vagrant$ npm start

> habitrpg@0.0.0-152 start /vagrant
> grunt run:dev

grunt-cli: The grunt command line interface. (v0.1.13)

Fatal error: Unable to find local grunt.

If you're seeing this message, either a Gruntfile wasn't found or grunt
hasn't been installed locally to your project. For more information about
installing and configuring grunt, please see the Getting Started guide:

http://gruntjs.com/getting-started

npm ERR! habitrpg@0.0.0-152 start: `grunt run:dev`
npm ERR! Exit status 99
npm ERR!
npm ERR! Failed at the habitrpg@0.0.0-152 start script.
npm ERR! This is most likely a problem with the habitrpg package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     grunt run:dev
npm ERR! You can get their info via:
npm ERR!     npm owner ls habitrpg
npm ERR! There is likely additional logging output above.
npm ERR! System Linux 3.2.0-23-generic
npm ERR! command "/usr/bin/node" "/usr/bin/npm" "start"
npm ERR! cwd /vagrant
npm ERR! node -v v0.10.26
npm ERR! npm -v 1.4.3
npm ERR! code ELIFECYCLE
npm ERR!
npm ERR! Additional logging details can be found in:
npm ERR!     /vagrant/npm-debug.log
npm ERR! not ok code 0

Why is it doing that? I thought the whole point of Vagrant was to eliminate issues like this.

**Comment by SergeantFTC** [comment]
And here's the results of "sudo npm install grunt"

npm http GET https://registry.npmjs.org/grunt
npm http 304 https://registry.npmjs.org/grunt
npm http GET https://registry.npmjs.org/colors
npm http GET https://registry.npmjs.org/async
npm http GET https://registry.npmjs.org/coffee-script
npm http GET https://registry.npmjs.org/dateformat/1.0.2-1.2.3
npm http GET https://registry.npmjs.org/eventemitter2
npm http GET https://registry.npmjs.org/findup-sync
npm http GET https://registry.npmjs.org/glob
npm http GET https://registry.npmjs.org/hooker
npm http GET https://registry.npmjs.org/iconv-lite
npm http GET https://registry.npmjs.org/minimatch
npm http GET https://registry.npmjs.org/nopt
npm http GET https://registry.npmjs.org/rimraf
npm http GET https://registry.npmjs.org/lodash
npm http GET https://registry.npmjs.org/underscore.string
npm http GET https://registry.npmjs.org/which
npm http GET https://registry.npmjs.org/js-yaml
npm http GET https://registry.npmjs.org/exit
npm http GET https://registry.npmjs.org/getobject
npm http GET https://registry.npmjs.org/grunt-legacy-util
npm http 304 https://registry.npmjs.org/colors
npm http 304 https://registry.npmjs.org/coffee-script
npm http 304 https://registry.npmjs.org/findup-sync
npm http 304 https://registry.npmjs.org/glob
npm http 304 https://registry.npmjs.org/hooker
npm http 304 https://registry.npmjs.org/iconv-lite
npm http 304 https://registry.npmjs.org/minimatch
npm http 304 https://registry.npmjs.org/rimraf
npm http 304 https://registry.npmjs.org/dateformat/1.0.2-1.2.3
npm http 304 https://registry.npmjs.org/nopt
npm http 304 https://registry.npmjs.org/eventemitter2
npm http 304 https://registry.npmjs.org/underscore.string
npm http 304 https://registry.npmjs.org/js-yaml
npm http 304 https://registry.npmjs.org/exit
npm http 304 https://registry.npmjs.org/getobject
npm http 304 https://registry.npmjs.org/which
npm http 304 https://registry.npmjs.org/grunt-legacy-util
npm http 304 https://registry.npmjs.org/lodash
npm http 304 https://registry.npmjs.org/async
npm ERR! error rolling back Error: UNKNOWN, unlink '/vagrant/node_modules/grunt/node_modules /colors/package.json'
npm ERR! error rolling back  grunt@0.4.4 { [Error: UNKNOWN, unlink '/vagrant/node_modules/gr unt/node_modules/colors/package.json']
npm ERR! error rolling back   errno: -1,
npm ERR! error rolling back   code: 'UNKNOWN',
npm ERR! error rolling back   path: '/vagrant/node_modules/grunt/node_modules/colors/package .json' }
npm ERR! Error: UNKNOWN, symlink '../which/bin/which'
npm ERR! If you need help, you may report this _entire_ log,
npm ERR! including the npm and node versions, at:
npm ERR!     http://github.com/npm/npm/issues

npm ERR! System Linux 3.2.0-23-generic
npm ERR! command "/usr/bin/node" "/usr/bin/npm" "install" "grunt"
npm ERR! cwd /vagrant
npm ERR! node -v v0.10.26
npm ERR! npm -v 1.4.3
npm ERR! path ../which/bin/which
npm ERR! code UNKNOWN
npm ERR! errno -1
npm ERR! Error: ENOENT, lstat '/vagrant/node_modules/grunt/node_modules/glob/examples/usr-lo cal.js'
npm ERR! If you need help, you may report this _entire_ log,
npm ERR! including the npm and node versions, at:
npm ERR!     http://github.com/npm/npm/issues

npm ERR! System Linux 3.2.0-23-generic
npm ERR! command "/usr/bin/node" "/usr/bin/npm" "install" "grunt"
npm ERR! cwd /vagrant
npm ERR! node -v v0.10.26
npm ERR! npm -v 1.4.3
npm ERR! path /vagrant/node_modules/grunt/node_modules/glob/examples/usr-local.js
npm ERR! fstream_path /vagrant/node_modules/grunt/node_modules/glob/examples/usr-local.js
npm ERR! fstream_type File
npm ERR! fstream_class FileWriter
npm ERR! code ENOENT
npm ERR! errno 34
npm ERR! fstream_stack /usr/lib/node_modules/npm/node_modules/fstream/lib/writer.js:284:26
npm ERR! fstream_stack Object.oncomplete (fs.js:107:15)
npm ERR! EEXIST, mkdir '/vagrant/node_modules/grunt/node_modules/js-yaml/examples'
File exists: /vagrant/node_modules/grunt/node_modules/js-yaml/examples
Move it away, and try again.

npm ERR! System Linux 3.2.0-23-generic
npm ERR! command "/usr/bin/node" "/usr/bin/npm" "install" "grunt"
npm ERR! cwd /vagrant
npm ERR! node -v v0.10.26
npm ERR! npm -v 1.4.3
npm ERR! path /vagrant/node_modules/grunt/node_modules/js-yaml/examples
npm ERR! fstream_path /vagrant/node_modules/grunt/node_modules/js-yaml/examples/dumper.json
npm ERR! fstream_type File
npm ERR! fstream_class FileWriter
npm ERR! code EEXIST
npm ERR! errno 47
npm ERR! fstream_stack /usr/lib/node_modules/npm/node_modules/fstream/lib/writer.js:171:23
npm ERR! fstream_stack /usr/lib/node_modules/npm/node_modules/mkdirp/index.js:37:53
npm ERR! fstream_stack Object.oncomplete (fs.js:107:15)
npm http GET https://registry.npmjs.org/abbrev
npm http 200 https://registry.npmjs.org/abbrev
npm http GET https://registry.npmjs.org/abbrev/-/abbrev-1.0.4.tgz
npm http 200 https://registry.npmjs.org/abbrev/-/abbrev-1.0.4.tgz
npm ERR!
npm ERR! Additional logging details can be found in:
npm ERR!     /vagrant/npm-debug.log
npm ERR! not ok code 0

**Comment by thepeopleseason** [comment]
Hmm.. I'll investigate.

**Comment by thepeopleseason** [comment]
Can you post the contents of your Vagrantfile?

**Comment by SergeantFTC** [comment]
sure.

```
# -*- mode: ruby -*-
# vi: set ft=ruby :

# The vagrant config for HabitRPG. Requires vagrant on your local machine.
# The box fetched will be precise64 located:
#    http://files.vagrantup.com/precise64.box

Vagrant.configure("2") do |config|
  config.vm.box = "habitrpg"
  config.vm.box_url = "http://dl.dropboxusercontent.com/u/4309797/devel/habitrpg/habitrpg.box"
  config.vm.hostname = "habitrpg"
  config.vm.network "forwarded_port", guest: 3000, host: 3000
  #config.vm.provision :shell, :path => "vagrant.sh"
end
```

**Comment by thepeopleseason** [comment]
Looks like everything I did to install grunt, bower, and habitrpg didn't take in the vagrantbox. To get up and running, you'll want to follow the [Unix instructions here](http://habitrpg.wikia.com/wiki/Setting_up_HabitRPG_locally) from line 6 forward (you'll need to sudo to install grunt-cli and bower, as the note says). I will update the vagrantbox.

Also, check for some instructions in #3061

**Comment by thepeopleseason** [comment]
I've updated the habitrpg.box file on my Dropbox, deleted all local vagrant images, and re-ran the vagrant install instructions, and successfully run npm start. 

If you haven't tried the Unix installation instructions, you can delete the local vagrant image and retry. What operating system are you using?

**Comment by thepeopleseason** [comment]
@SergeantFTC Are you still having issues with the vagrant setup?

**Comment by SergeantFTC** [comment]
oh, sorry. I tried your updated box, and still couldn't get it working, so I just did it from scratch on a new Ubuntu Server vm, which is now working. Thanks :)

**Comment by thepeopleseason** [comment]
We should be able to close this issue now (fixed by #3256).
